src/pages/index.js

```diff
@@ -32,71 +87,145 @@ const cardPopOut = new PopupWithImage({
 
 cardPopOut.setEventListeners();
 
-function createCard(item) {
-  const card = new Card(item, "#localeCard", openModalImage);
-  return card.generateCard();
+const deleteCardModal = new PopupWithDelete({
+  popupSelector: ".modal_type_delete-image",
+  handleBtnClick: () => {
+    api.deleteCard(cardToDelete.data._id);
+    cardToDelete.deleteCard();
+    deleteCardModal.close();
+  },
+});
+
+/*.then(cardToDelete.deleteCard())*/
+
+function handleDeleteClick(card) {
+  console.log(card);
+  deleteCardModal.open();
+  deleteCardModal.setEventListeners();
+  cardToDelete = card;
+  return cardToDelete;
+  /*
+  return card;
+  cardToDelete = card;
+  console.log(cardToDelete);
+  */
 }
 
-const firstCards = new Section(
-  { items: initialCards, renderer: createCard },
-  ".elements"
-);
+function handleLike() {
+  if (this.data.isLiked == false) {
+    api.addLike(this.data._id);
+    this.data.isLiked = true;
+  } else {
+    api.removeLike(this.data._id);
+    this.data.isLiked = false;
+  }
+}
 
-firstCards.renderItems();
+//Handle edit/add button clicks
 
-const newUserInfo = new UserInfo({
-  userNameSelector: ".info__name",
-  userJobSelector: ".info__job-title",
+modalProfileEditBtn.addEventListener("click", () => {
+  profileModal.open();
+  const data = newUserInfo.getUserInfo();
+  console.log(data);
+  modalName.value = data.name;
+  modalJob.value = data.job;
+  profileValidator.resetValidation();
+});
+
+newProfileImageBtn.addEventListener("click", () => {
+  newProfilePictureModal.open();
+});
+
+modalImageEditBtn.addEventListener("click", () => {
+  newImageModal.open();
 });
 
 const profileModal = new PopupWithForm({
   popupSelector: ".modal_type_profile",
   handleFormSubmit: (formData) => {
-    console.log(formData.name);
-    console.log(formData.job);
-    newUserInfo.setUserInfo({ name: formData.name, job: formData.job });
+    /*
+    profileModal.renderSaving(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```